app/src/main/java/com/parasoft/demoapp/component/OrderDialog.java

```diff
@@ -157,7 +160,13 @@ private void closeAndRefresh() {
     }
 
     private void getOrderDetails() {
-        pdaService.getClient(ApiInterface.class).orderDetails(orderNumber)
+        if(Looper.myLooper() == null) {
```

Comment:
Please remove this block of code and set "Save“ button disabled as default in the layout xml file.
And enable "Save“ button when valid option selected and triggered "setOnItemSelectedListener"